### PR TITLE
remove volume average msa and allow dynamic toll from initial iteration

### PR DESCRIPTION
**Pull request by ychtw**

### CODE REVIEW

File: tm2py/components/network/highway/highway_assign.py [hunk 1]
```diff
@@ -152,17 +152,16 @@ def run(self):
                     None
                 self._create_skim_matrices(scenario, assign_classes)
 
-                if iteration == 0 or not run_dynamic_toll:
-                    assign_spec = self._get_assignment_spec(assign_classes)
+                assign_spec = self._get_assignment_spec(assign_classes)
+
+                if not run_dynamic_toll:
```
Comment (i-am-sijia) [inline tm2py/components/network/highway/highway_assign.py:155]: Missing condition: If warmstart is turned off, iteration 0 does not run dynamic tolling.

**Review by i-am-sijia — CHANGES_REQUESTED** [review]
Missing a condition for dynamic tolling.

File: tm2py/components/network/highway/highway_assign.py (hunk 1 above)
Comment (ychtw) [reply tm2py/components/network/highway/highway_assign.py:155]: updated in 178cd28

File: tm2py/components/network/highway/highway_assign.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def run(self):
         run_dynamic_toll = self.config.tolls.run_dynamic_toll
         valuetoll_start_tollbooth_code = self.config.tolls.valuetoll_start_tollbooth_code
         max_dynamic_valuetoll = self.config.tolls.max_dynamic_valuetoll
-        apply_msa = self.config.msa.apply_msa
+        warmstart = self.config.run.warmstart.warmstart
```
Comment (i-am-sijia) [inline tm2py/components/network/highway/highway_assign.py]: This does not look correct. Shouldn't it be `self.controller.config.run.warmstart.warmstart`?

Comment (ychtw) [reply tm2py/components/network/highway/highway_assign.py]: oh, I have this fixed, but did not push it up here. updated in 7921a19

**Review by i-am-sijia — APPROVED** [review]

**Pull request merged**